src/XR.tsx

```diff
@@ -204,10 +198,12 @@ function XR({
   React.useEffect(() => {
     if (!session) return
 
-    const handleSessionStart = (nativeEvent: XRManagerEvent) => onSessionStart?.({ nativeEvent, session })
-    const handleSessionEnd = (nativeEvent: XRManagerEvent) => onSessionEnd?.({ nativeEvent, session })
-    const handleVisibilityChange = (nativeEvent: XRSessionEvent) => onVisibilityChange?.({ nativeEvent, session })
-    const handleInputSourcesChange = (nativeEvent: XRSessionEvent) => onInputSourcesChange?.({ nativeEvent, session })
+    const handleSessionStart = (nativeEvent: XRManagerEvent) =>
+      onSessionStart?.({ nativeEvent: { ...nativeEvent, target: session }, target: session })
+    const handleSessionEnd = (nativeEvent: XRManagerEvent) =>
+      onSessionEnd?.({ nativeEvent: { ...nativeEvent, target: session }, target: session })
+    const handleVisibilityChange = (nativeEvent: XRSessionEvent) => onVisibilityChange?.({ nativeEvent, target: session })
+    const handleInputSourcesChange = (nativeEvent: XRSessionEvent) => onInputSourcesChange?.({ nativeEvent, target: session })
```

Comment:
Three's dispatcher types will specify `any` for sessionstart and sessionend, but it's actually an instance of the `THREE.WebXRManager`. I'll overwrite them here to align the events with `XRSessionEvent`.